### ISSUE TITLE
tsgo fails to parse .svelte.ts files with Svelte 5 runes ($state, $derived, etc.)

### ISSUE THREAD

**Issue by chbert** [pheuter/svelte-check-rs#35]
## Description

When using `svelte-check-rs` on a codebase with `.svelte.ts` files that use Svelte 5 runes (`$state`, `$derived`, etc.), tsgo reports hundreds of parse errors because it doesn't understand the runes syntax.

## Reproduction

**File: `src/lib/stores/example.svelte.ts`**
```typescript
import { SvelteMap } from 'svelte/reactivity';

class MyStore {
  // This causes: Error: ')' expected. (ts(TS1005))
  eventOrder = $state<'option-a' | 'option-b'>('option-a');
  
  // This also fails
  items = $state<string[]>([]);
  
  // SvelteMap with $state also fails
  cache = new SvelteMap<string, number>();
}

export const store = new MyStore();
```

**Command:**
```bash
svelte-check-rs --tsconfig ./tsconfig.json
```

**Result:**
```
src/lib/stores/example.svelte.ts:5:44
Error: ')' expected. (ts(TS1005))

src/lib/stores/example.svelte.ts:5:65
Error: ',' expected. (ts(TS1005))
...
```

## Impact

In our production SvelteKit codebase:
- **svelte-check**: 1 error (a real type error)
- **svelte-check-rs**: 584 errors (mostly false positives from .svelte.ts files)

Breakdown:
- 476 errors in `.svelte.ts` files (tsgo can't parse runes)
- 108 errors in `.svelte` files (mix of real errors and transform issues)

## Expected Behavior

`.svelte.ts` files should be transformed similarly to `.svelte` files to convert runes syntax into valid TypeScript before being passed to tsgo.

## Workaround

Using `--skip-tsgo` reduces errors to just 9 (Svelte parse errors) + 29 warnings, but this loses TypeScript type-checking entirely.

## Environment

- svelte-check-rs: 0.5.2
- tsgo: 7.0.0-dev.20260106.1
- Svelte: 5.45.2
- SvelteKit: 2.48.0
- macOS (Apple Silicon)

## Additional Context

The `.svelte.ts` extension is the standard convention for Svelte 5 reactive modules that use runes outside of `.svelte` components. See: https://svelte.dev/docs/svelte/svelte-files#.svelte.ts

**Comment by pheuter** [comment]
svelte-check-rs has support and test coverage for svelte-modules: https://github.com/pheuter/svelte-check-rs/blob/main/test-fixtures/projects/svelte-modules/src/lib/counter-state.svelte.ts

if you can produce a minimal repro, that would be really helpful for me to investigate. Thanks.
